File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int i=0;
        int j=n-1;
        int max=-1;
        while(i<=j){
            max=Math.max(max,(Math.min(height[i],height[j])*(j-i)));
            if(height[i]<height[j]) i++;
            else j--;
        }
        return max;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1

        val=-1
        while i<=j:
            val=max(val,min(height[i],height[j])*(j-i))
            if height[i]<height[j]:
                i+=1
            else:
                j-=1

        return val


        